rgo/builders/copr.py

```diff
@@ -161,19 +164,44 @@ def build(self, component, with_build_id=None, after_build_id=None):
             "after_build_id": after_build_id,
         }
 
-        if self.chroots:
-            buildopts["chroots"] = self.chroots
-
-        build = self.client.build_proxy.create_from_file(self.owner, self.name, component.srpm, buildopts=buildopts)
-        LOGGER.info('Submitted Copr build for %s chroots: %s %s%sURL: %s' % (
-            component.name,
-            ", ".join(buildopts["chroots"]) if "chroots" in buildopts else "(all)",
-            "with: %s " % with_build_id if with_build_id is not None else "",
-            "after: %s " % after_build_id if after_build_id is not None else "",
-            self._build_url(build.id)
-        ))
+        if self.cmdline_input_chroots:
+            buildopts["chroots"] = self.cmdline_input_chroots
+
+        overriden_chroots = set()
+        for override in component.overrides:
+            overriden_chroots.update(override.roots)
+
+        # if we have atleast one default (not overriden) chroot do the default build
+        if (list(self.all_project_chroots - overriden_chroots)):
+            buildopts["chroots"] = list(self.all_project_chroots - overriden_chroots)
+            build = self.client.build_proxy.create_from_file(self.owner,
+                                                             self.name,
+                                                             component.srpm,
+                                                             buildopts=buildopts)
+            LOGGER.info('Submitted Copr build for %s chroots: %s %s%sURL: %s' % (
+                component.name, ", ".join(buildopts["chroots"]),
+                "with: %s " % with_build_id if with_build_id is not None else "",
+                "after: %s " % after_build_id if after_build_id is not None else "",
+                self._build_url(build.id)
+            ))
+            default_build_id = build.id
+
+        for override in component.overrides:
+            if override.roots:
+                buildopts["chroots"] = list(override.roots)
```

Comment:
And here this unconditionally builds for all overridden chroots, which:
- Is not what is expected; I think what's expected is to build either `--chroots` specified on CLI or all Copr project chroots. Any overrides that are not on that list should just be ignored for this build (the overlay can be used in multiple places, you don't always want to build the chroots that have overrides)
- Will IMO break if you've got overrides for chroots that aren't configured in the Copr project.

Reply:
Should be hopefully working as expected now.